app/assets/javascripts/discourse/app/controllers/change-post-notice.js

```diff
@@ -1,58 +0,0 @@
-import Controller from "@ember/controller";
-import ModalFunctionality from "discourse/mixins/modal-functionality";
-import { action } from "@ember/object";
-import { cookAsync } from "discourse/lib/text";
-import discourseComputed from "discourse-common/utils/decorators";
-import { isEmpty } from "@ember/utils";
-
-export default Controller.extend(ModalFunctionality, {
-  post: null,
-  resolve: null,
-  reject: null,
-
-  notice: null,
-  saving: false,
-
-  @discourseComputed("saving", "notice")
-  disabled(saving, notice) {
-    return saving || isEmpty(notice);
-  },
-
-  onShow() {
-    this.setProperties({ notice: "", saving: false });
-  },
-
-  onClose() {
-    if (this.reject) {
-      this.reject();
-    }
-  },
```

Comment:
Looks like the old implementation would call `.reject()` when the modal is closed (e.g. via the X button in the header). Do we need to replicate that by creating a wrapper around `@closeModal` in the component? (or maybe it's not really needed at all... is anything actually listening for the promise to be rejected?)